torch/_export/utils.py

```diff
@@ -45,6 +58,106 @@
 }
 
 
+# Our strategy for deciding if we can preserve a op is following:
+# 1. The op should be known statically that it is functional
+# 2. If it is maybe aliasing, we decompose because we must know if an op
+#    is mutating or aliasing.
+# TODO (tmanlaibaatar) make this utility function and share it with functional_tensor
+# decomp part. (https://github.com/pytorch/pytorch/issues/129431)
+def _assert_valid_to_preserve(op_overload):
+    if op_overload in FunctionalTensor.maybe_aliasing_or_mutating_ops:
```

Comment:
(you will still need to skip CIA ops that are mutable and maybe-aliasing though when deciding which ops to override with `NotImplemented`)

Reply:
Can you elaborate more? We already skip those CIA ops.

Reply:
Oh what I mean is:

(1) you have some logic in the PR that will need to loop over every CIA op in the dispatcher, and populate an entry in the `decomp_table`, e.g. `{op: not_implemented_fn}`.

(2) but you will also need to carefully make sure that you do **not** do this for the "mutating or maybe-aliasing CIA ops", since we still want those decomps to run